packages/itwinui-css/src/input/searchbox.scss

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@mixin iui-expandable-searchbox {
+  overflow: hidden;
+  width: var(--_iui-input-min-height);
+  transition: width var(--iui-duration-3) ease;
+
+  &::before {
+    content: '';
+    position: absolute;
+    inset: 0;
+    background-color: inherit;
+    z-index: 3;
+    pointer-events: none;
+  }
+
+  &::after {
+    --search-svg: url('data:image/svg+xml,<svg viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg">\
+      <path d="M11 9.7c.7-1 1.1-2.2 1.1-3.5.1-3.5-2.7-6.2-6-6.2C2.7 0 0 2.7 0 6.1s2.7 6.1 6.1 6.1c1.3 0 2.5-.4 3.5-1.1l4.9 4.9 1.4-1.4zm-5 .5c-2.3 0-4.1-1.8-4.1-4.1S3.7 2 6 2s4.1 1.8 4.1 4.1-1.8 4.1-4.1 4.1"/>\
+    </svg>');
+
+    content: '';
+    position: absolute;
+    left: calc(var(--_iui-input-min-height) / 2);
+    top: 50%;
+    transform: translate(-50%, -50%);
+    background-color: currentColor;
+    mask: var(--search-svg) no-repeat center / contain;
+    width: 1rem;
+    height: 1rem;
+    pointer-events: none;
+    cursor: pointer;
+    z-index: 3;
+  }
+
+  &[data-iui-expanded='true'],
+  &:focus-within {
+    width: 100%;
+    border-color: var(--_iui-input-border-color);
+
+    &::before {
+      background: unset;
+    }
+
+    &::after {
+      content: unset;
+    }
+  }
+}
+
+@mixin iui-search-icon {
+  align-self: center;
+  margin: 0 var(--iui-size-s);
```

Comment:
this should probably be padding instead of margin, and the `align-self` can be handled in parent (using `align-items`)

Reply:
Did those changes

Reply:
it's still using margin